### PR TITLE
[REVIEW] Fix to a scalar datetime(of type Days) issue

### PR DESCRIPTION
**Pull request by galipremsagar**
This fixes:
1. #2383 
2. Build issue

We seem to be typecasting all datetimes to datetime[ms] in our series column formation too. Performing a similar typecast for a scalar value aswell.

### CODE REVIEW

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
=== 11558 passed, 759 skipped, 85 xfailed, 6 xpassed, 1903 warnings in 376.45 seconds ===

File: python/cudf/cudf/dataframe/dataframe.py [hunk 1]
```diff
@@ -1141,6 +1141,15 @@ def _sanitize_values(self, series, SCALAR):
...
         """
         if SCALAR:
+            if hasattr(series, "dtype") and (
+                series.dtype
+                in (
+                    np.dtype("datetime64[D]"),
+                    np.dtype("datetime64[M]"),
+                    np.dtype("datetime64[Y]"),
+                )
+            ):
```
Comment (kkraus14) [inline python/cudf/cudf/dataframe/dataframe.py]: Instead of doing this should we just use `np.issubdtype(series.dtype, np.datetime64)`?

Comment (galipremsagar) [reply python/cudf/cudf/dataframe/dataframe.py]: Yes, changed to subdtype check

`= 11558 passed, 759 skipped, 85 xfailed, 6 xpassed, 1903 warnings in 377.46 seconds =`

**Pull request merged**